collections/_tutorials/2022-09-12-tutorial-cvae.md

```diff
@@ -0,0 +1,296 @@
+
+### ELBO reformulation
+
+The ELBO $$\mathcal{L}$$ should be reformulated so to justify the loss involved in the conditional VAE framework. The corresponding derivation is provided below.
+
+$$\mathcal{L} = \int{q(z/x,y) \cdot log\left(\frac{p(y,z/x)}{q(z/x,y)}\right) \,dz}$$
+
+By using the following ***conditional probability*** relations:
+
+$$
+p(x,y,z) = \left\{
+  \begin{array}
+    pp(y,z/x) \cdot p(x) \\
+    p(y/x,z) \cdot p(x,z)
+  \end{array}
+  \right.
+$$
+
+$$p(x,y) = p(z/x) \cdot p(x)$$
+
+the next equations can be easily obtained
+
+$$p(y,z/x) = \frac{p(y/x,z) \cdot p(x,z)}{p(x)}$$
+
+$$p(y,z/x) = \frac{p(y/x,z) \cdot p(z/x) \cdot p(x)}{p(x)}$$
+
+$$p(y,z/x) = p(y/x,z) \cdot p(z/x)$$
+
+&nbsp;
+
+The ELBO $$\mathcal{L}$$ expression can thus be rewritten as
+
+$$\mathcal{L} = \int{q(z/x,y) \cdot log\left(\frac{p(y/x,z) \cdot p(z/x)}{q(z/x,y)}\right) \,dz}$$
+
+$$\mathcal{L} = \int{q(z/x,y) \cdot \left[log\left(p(y/x,z)\right) + log\left(\frac{p(z/x)}{q(z/x,y)}\right) \right] \,dz}$$
+
+$$\mathcal{L} = \int{q(z/x,y) \cdot log\left(p(y/x,z)\right) \,dz} \,+\, \int{q(z/x,y) \cdot log\left(\frac{p(z/x)}{q(z/x,y)}\right) \,dz}$$
+
+$$\mathcal{L} =  \mathbb{E}_{z\sim q(z/x,y)} \left[log\left(p(y/x,z)\right)\right] - D_{KL}\left(q(z/x,y)\parallel p(z/x)\right)$$
+
+where $$\mathbb{E}_{z\sim q(z/x,y)}$$ is the mathematical expectation with respect to $$q(z/x,y)$$. 
+
+&nbsp;
+
+At this stage of analysis, it is necessary to model $$p(y/x,y)$$ via an ***a priori distribution***. Let's first note that $$p(y/x,z)$$ is modeled by a neural network $$f(\cdot)$$ so that $$\hat{y}=f(x,z)$$. Since this function is deterministic, it will allow to model $$p\left(y/\hat{y}\right)$$. By approximating $$p\left(y/\hat{y}\right)$$ by a Bernoulli distribution, we have
+
+$$\mathbb{E}_{z\sim q(z/x,y)} \left[log\left(p(y/\hat{y})\right)\right] = \mathbb{E}_{z\sim q(z/x,y)} \left[log\left( {\hat{y}}^y \cdot \left(1-\hat{y}\right)^{1-y} \right)\right]$$
+
+$$ = \mathbb{E}_{z\sim q(z/x,y)} \left[ y \, log\left( \hat{y} \right) + \left(1-y\right) \, log\left(1-\hat{y}\right) \right]$$
+
+$$ = \mathbb{E}_{z\sim q(z/x,y)} \left[-CE\left( y,f\left(x,z\right)\right)\right]$$
+
+where $$CE(\cdot)$$ corresponds to the conventional ***cross entropy function*** !
+
+&nbsp;
+
+Thanks to this modeling, we are finally looking for:
+
+$$\left(f^*,g^*,h^*,k^*,l^*\right) = \underset{(f,g,h,k,l)}{\arg\max} \,\,\, \left( \mathbb{E}_{z\sim q(z/x,y)} [-CE\left( y,f\left(x,z\right)\right)] - D_{KL}(\underbrace{q(z/x,y)}_{g,h}\parallel \underbrace{p(z/x)}_{k,l}) \right)$$
+
+$$\left(f^*,g^*,h^*,k^*,l^*\right) = \underset{(f,g,h,k,l)}{\arg\min} \,\,\, \left( \mathbb{E}_{z\sim q(z/x,y)} [CE\left( y,f\left(x,z\right)\right)] + D_{KL}(\underbrace{q(z/x,y)}_{g,h}\parallel \underbrace{p(z/x)}_{k,l}) \right)$$
+
+&nbsp;
+
+## Various scenarios
+
+There are different exploitations of the cVAE formalism depending on the prior modeling $$p(z/x)$$ and the content of the $$y$$ references.
```

Comment:
_exploitations_ is not a word in English (at least that I know of). But I'm not entirely sure what you meant by it, so I'm just gonna give you 1-2 suggestions here and let you decide what you like best based on what you meant:

- _ways to leverage_ the cVAE [...]?
- _methods based on_ the cVAE [...]?

Reply:
thanks ;) I chose the first one ;)